osquery/tables/applications/linux/namespace_ops.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <fcntl.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+
+#include <osquery/tables/applications/linux/namespace_ops.h>
+
+namespace osquery {
+namespace tables {
+
+Status NamespaceOps::invoke(std::function<void(int fd)> fn) {
+  std::string pidns = "/proc/" + std::to_string(_pid) + "/ns/mnt";
+  int fd = open(pidns.c_str(), O_RDONLY);
+  if (fd == -1) {
+    return Status(1, strerror(errno));
+  }
+
+  pid_t childPid = fork();
+  if (childPid == -1) {
+    int fErrno = errno;
+    close(fd);
+    return Status(1, strerror(fErrno));
+  }
+  if (childPid != 0) {
+    _childPid = childPid;
+    close(fd);
+    return Status();
+  }
+
+  // Change the namespace
+  if (setns(fd, 0) == -1) {
+    close(fd);
+    close(_sockFD);
+    _Exit(EXIT_SUCCESS);
+  }
+
+  // Execute the function
+  fn(_sockFD);
+
+  // Close the file handles and exit
+  close(fd);
+  close(_sockFD);
+  _Exit(EXIT_SUCCESS);
+}
+
+Status NamespaceOps::wait() {
+  siginfo_t infop;
+  while (1) {
```

Comment:
Should this perhaps be bound by some timeout as well as the `continue` below?

Reply:
Continuing off of what @marpaia is mentioning, and forgive me if I'm missing something, but I am also not seeing how you'd ever break out of this? Does this code not infinite loop? `waitid` will return your success, sure, but it seems that your only way out of this loop is if `waitid` errors.

Reply:
@muffins That's a good catch. The reason why this was working, was because the logic was working - after waitid returns success and we made it wait again, it would return an error because that pid was collected by the kernel. Fixed.

Reply:
@diptanu, It looks like we're still waiting for a response or change to address Mike's comment about a potential infinite loop vs. a timeout.